doxygen: Move doxygen files into docs

jdemel opened the pull request:
This is a first stab at re-arranging doxygen files to have fewer files in our top-level directory.

This includes all `*.md` files as well. We might want to exclude some of them.

jdemel committed:
doxygen: Move doxygen files into docs

This is a first stab at re-arranging doxygen files to have fewer files
in our top-level directory.

Signed-off-by: Johannes Demel <demel@uni-bremen.de>

michaelld commented on CMakeLists.txt:
```diff
@@ -209,16 +209,7 @@ endif(ENABLE_ORC)
-        WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-        COMMENT "Generating documentation with Doxygen" VERBATIM
-    )
+  add_subdirectory(docs)
```
Let's move this whole section, from `find_package(Doxygen)` to `endif(DOXYGEN_FOUND)` into the subdirectory. That's better CMake programming: only check in the current CMakeLists.txt file for stuff that's required within it, and don't rely on previous checks that may or not have taken place.

jdemel replied:
I agree. Are there actually any resources that teach you "good CMake"?

michaelld replied:
Hmmm ... good question ... IDK .. mine is via experience, having used CMake now since GR switched, ages ago; and, working on tons or projects that use CMake

jdemel replied:
I moved it.

michaelld reviewed (changes requested):
Minor tweak & this will be good to go. I need to learn how to edit PRs ... not tonight though

jdemel commented:
Should some of the `*.md` files be excluded?

michaelld commented:
> Should some of the `*.md` files be excluded?

Not sure I follow this comment for this PR? Maybe you meant #580 ?

jdemel commented:
If we'd merge #580 (and I assume we do now), then my comment is more relevant. I'll wait for the other things to happen and update this PR accordingly. I assume I mixed up some files.

michaelld commented:
OK #580 is merged ... once this PR's code is tweaked per my request then I'm good with it. @jdemel if there are other changes needed post-580 then please go ahead & push commits for them for us to review.

jdemel committed:
doxygen: Move Doxygen package check into docs

Now, all the checks and code for our `docs` subfolder are in one place.

Signed-off-by: Johannes Demel <demel@uni-bremen.de>

jdemel commented:
Some `.md` files that I consider to be clutter are included in the docs at the moment.

I feel like these should not be included:
- top-level `README.md`
- docs/AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md`

e.g. `docs/versioning.md` were not part of our docs build before. But I assume these fit nicely in the docs as well.

michaelld reviewed (approved):
LGTM now; thx!

michaelld commented:
The top-level `README.md` is required for best GitHub presentation, so that needs to stay there. I'm OK with `AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md` and `versioning.md` being in `docs` since they fit there. Once this PR is merged, I think we're at a minimum number of extraneous files at the top-level ... do you agree?

michaelld commented:
I mean those extra `.md` files in `docs` are just along for the ride; we don't build them into the actual Doxygen docs beyond links, do we?

michaelld commented:
If you feel some of those files need to be moved & it makes sense to put them into this PR to go for it. Otherwise, I'd say let's merge this one & get the next round going.

jdemel commented:
I want these files to stay where they are. I just don't want to integrate them into the Doxygen build. I'll exclude them.

jdemel committed:
doxygen: Exclude unnecessary files

The top-level `README.md` and
`docs/AUTHORS_RESUBMITTING_UNDER_LGPL_LICENSE.md` files are now excluded
from the doxygen build.

Signed-off-by: Johannes Demel <demel@uni-bremen.de>

jdemel commented:
Now, this PR should be ready to be merged. @michaelld I leave this to you since it is my PR.

michaelld reviewed (approved):
LGTM ; merging
